src/workers/parser/mapBindings.js

```diff
@@ -62,6 +135,11 @@ export default function mapExpressionBindings(
   let shouldUpdate = true;
 
   t.traverse(ast, (node, ancestors) => {
+    // the isMapped check is to avoid processing the newly added nodes
+    if (isMapped) {
```

Comment:
This isn't a safe way to bail because we may well want to perform more changes, even if part of the AST has been mapped. Ideally the newly-added nodes would be simplified enough that they wouldn't trigger further processing. Do you have example of what this is trying to avoid?

Reply:
Well, basically in my case it covered the new declarations, such as `let __decl{index}__` and transformed it to `self.__decl{index}__`.
I applied the suggested changes and that check is not required anymore.